### DISCUSSION TITLE
Messages

### DISCUSSION THREAD

**Discussion by rvoortman** [etsy/open-api#677] [Ideas]
Is it possible to read and write message with the new API? I can't really find it in the docs.

This could be useful to have integrations into a crm system or support system.

**Comment by Gareth064** [comment]
Quick answer is No.

Other have raised this in the past under the following issues
#31 
#67 
#171 
#284

**Comment by wscourge** [comment]
Hey @Gareth064, links to those issues are no longer available. Do you happen to remember the reasoning behind not supporting it? I would like to write an anti-spam AI system for Etsy, but it is a no-go without messages CRUD API.

Thanks in advance.

**Reply by Gareth064** [reply]
They never gave a reason. They just said it won't happen.

I could speculate though. They don't want to expose too much functionality via the API which would reduce the reliance for shop owners to log into and use the Etsy Apps.

So this is a good one to keep you hooked into having to use their app to communicate with customers.

I think in the old API there was a way to expose the Etsy Checkout experience to other websites using the API, but again that was removed because they want everyone using their Apps as much as possible.

It's fair in one sense, but when developers like us are used to working with other Marketplaces or e-commerce platforms that have awesome and open APIs, it is frustrating when Etsy won't provide a similar API experience.

**Reply by wscourge** [reply]
I see, that's a shame. Thanks for taking a time to answer.

**Comment by davidbernat** [comment]
@Gareth064 @wscourge This is all very, very concerning coming from Etsy, and yet again implies we MUST DISBAND from Etsy.

1. Why are GitHub tickets being removed? I can legitimately think of no other GitHub repository in which I have seen that behavior.
2. Why would Etsy be so concerned with bot security that they do not even allow a user to READ their existing messages? Am I wrong?

Here is a simple app any story manager wants: remind me whenever I have a message unanswered for three days. Done. 
I did a few minute search online, and found that Etsy still has no intentions to implement these simply features? 

I am insulted.
Add this to the list of Google Login abuses and Personal API Key refusals.

**Reply by Gareth064** [reply]
Totally agree with you @davidbernat , it is frustrating that we don't have access to as much functionality, via Etsy API, as we do on other marketplaces.

Regarding the removing of tickets, this was in the very early days of this repo when they were developing v3 in the open with us as the testers. Once they reached GA with the v3 API, they removed Issues completely from the Repo and turned on Discussion only. Some issues that were still opened were converted into a discussion, but which issues were moved over and which were deleted was not something we had any say in.

**Comment by talk-fly** [comment]
Having buyer messages via API is a must these days. It what makes this solution scalable.

**Comment by davidbernat** [comment]
I agree @talk-fly. To be transparent to the open web, I do not really take Etsy seriously as a company. It has not provided adequate basic customer service, repeatedly, and which would be necessary to make up for basic lacks of features and outcome transparency. 

I do not know. I am a mostly incompetent market growth or purchase acquisition master, and struggle impressively both to create an online following as much as to funnel and trigger purchase decisions from what traffic I do receive. But I know automated systems, and how to predict and analyze what market funnel data I do have, and make product decisions. It just has not been a good experience. It lost the two year window we had before AI and A/B testing become the only dominant way to market analyze the product surface. In a different world I probably would love to have spent these two years working for a moderately small e-Commerce platform or store.
